src/Collection/Collection.router.ts

```diff
@@ -375,10 +379,11 @@ export class CollectionRouter extends Router {
 
   deleteCollection = async (req: AuthRequest) => {
     const id = server.extractFromReq(req, 'id')
+    const service = new CollectionService()
 
-    if (await this.isCollectionPublished(id)) {
+    if (await service.isLocked(id)) {
       throw new HTTPError(
-        "The collection is published. It can't be deleted",
+        `Cannot access the Collection ${id} because it's either locked or published`,
         { id },
         STATUS_CODES.unauthorized
```

Comment:
I wouldn't mix both errors, as it can be used to represent two different errors that we would like to show the user.
Also, the collection locked error (and maybe the error related to the collection being already published) should return a 423 (locked) instead of a 401 that relates to a user not being authenticated.